### DISCUSSION TITLE
联网搜索请求失败

### DISCUSSION THREAD

**Discussion by spongxin** [coaidev/coai#289] [Q&A | 问题解答] [answered]
[//]: # (方框内删除已有的空格，填 x 号)
+ [x] 我已确认目前没有类似 issue
+ [x] 我已确认我已升级到最新版本
+ [x] 我已完整浏览项目 README 和项目文档并未找到解决方案
+ [x] 我理解并愿意跟进此 issue，协助测试和提供反馈
+ [x] 我将以礼貌和尊重的态度提问，不得使用不文明用语 (包括在此发布评论的所有人同样适用, 不遵守的人将被 block)
+ [x] 我理解并认可上述内容，并理解项目维护者精力有限，**不遵循规则的 issue 可能会被无视或直接关闭**

**问题描述**
后台管理中搜索接入点填入有效的searXNG搜索节点（服务器docker部署），并且能够ping通；搜索引擎选择bing，
![image](https://github.com/user-attachments/assets/3019c987-e575-406b-b62d-6b57273c4c51)

**复现步骤**

**预期结果**

**日志信息**

**相关截图 (如果有)**
![image](https://github.com/user-attachments/assets/7487bbcd-65db-408e-b1bf-8d6f100840e9)

[a comment by an automated account is not shown]

**Comment by addy163** [comment]
绑定域名，开启ssl

[a comment by an automated account is not shown]

**Comment by zmh-program** [comment]
connect refuse，无法连接你的searxng服务

[a comment by an automated account is not shown]

**Comment by spongxin** [comment]
我为searxng服务以及coai服务全部绑定了域名并使用ssl，访问网站实测能够正常搜索，但是在coai后台仍然出现搜索错误：invalid character '<" looking for beginning of value，如图：
![image](https://github.com/user-attachments/assets/445a0b55-d4f7-4841-a47c-a02d83a2100e)

![image](https://github.com/user-attachments/assets/1fd406e1-5064-44a7-82f3-b3dc7ebef2c8)
searxng与coai部署在同一台机器，配置中填写了正确的完整域名

**Comment by SpiralTower** [comment, marked as answer]
通常情况下，searxNG不允许API调用，也就是说输出格式是HTML格式，你需要修改settings.yml来开启。
如果你使用docker部署，检查环境变量里是否有
`SEARXNG_SETTINGS_PATH=/etc/searxng/settings.yml` ，
如果有，请映射该配置文件到宿主机合适目录并在.yml中添加`- json` 。
```
search:
    formats:
         - html
         - json
```
重启后你可以使用`curl -kLX GET --data-urlencode q='searxNG' -d format=json http://你的域名`来检查是否返回JSON 。

**Reply by spongxin** [reply]
这非常有效！谢谢你现在我的coai能够顺利调用searxng服务了。

[a comment by an automated account is not shown]

**Comment by zmh-program** [comment]
> 我为searxng服务以及coai服务全部绑定了域名并使用ssl，访问网站实测能够正常搜索，但是在coai后台仍然出现搜索错误：invalid character '<" looking for beginning of value，如图：
> ![image](https://github.com/user-attachments/assets/445a0b55-d4f7-4841-a47c-a02d83a2100e)
> 
> ![image](https://github.com/user-attachments/assets/1fd406e1-5064-44a7-82f3-b3dc7ebef2c8)
> searxng与coai部署在同一台机器，配置中填写了正确的完整域名
> 

@spongxin 已经说的很好了，可以按照他的步骤来。

如果使用Docker，可以辅助参考下模板： http://github.com/zmh-program/searxng

**Comment by PFNOW** [comment]
我把localhost改成http://127.0.0.1，就可以用了
